pymatgen/io/aims/sets/base.py

```diff
@@ -0,0 +1,523 @@
+GEOMETRY_FILE_NAME: str = "geometry.in"
+
+
+DEFAULT_AIMS_PROPERTIES = ("energy", "free_energy", "forces", "stress", "stresses", "dipole", "magmom")
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class AimsInputFile(InputFile):
+    """The input file for an FHI-aims calculation.
+
+    Args:
+        _content_str (str): The contents of the input file as a string
+    """
+
+    content_str: str = ""
+
+    def get_str(self) -> str:
+        """Get the contents of the input file.
+
+        Returns:
+            str: The contents of the input file
+        """
+        return self.content_str
+
+    @classmethod
+    def from_str(cls, contents: str):
+        """Create an input file from the contents string.
+
+        Args:
+            contents (str): The contents of the input file
+        """
+        return cls(contents)
+
+
+class AimsInputSet(InputSet):
+    """A class to represent a set of Aims inputs."""
+
+    def __init__(
+        self,
+        parameters: dict[str, Any],
+        structure: Structure | Molecule,
+        properties: Sequence[str] = ("energy", "free_energy"),
+    ) -> None:
+        """Construct the AimsInputSet.
+
+        Args:
+            parameters (dict[str, Any]): The ASE parameters object for the calculation
+            structure (Structure or Molecule): The Structure/Molecule objects to
+                create the inputs for
+            properties (Sequence[str]): The properties to calculate for the calculation
+        """
+        self._parameters = parameters
+        self._structure = structure
+        self._properties = properties
+
+        aims_control_in, aims_geometry_in = self.get_input_files()
+        super().__init__(
+            inputs={
+                CONTROL_FILE_NAME: aims_control_in,
+                GEOMETRY_FILE_NAME: aims_geometry_in,
+                PARAMS_JSON_FILE_NAME: json.dumps(self._parameters, cls=MontyEncoder),
+            }
+        )
+
+    def get_input_files(self) -> tuple[str, str]:
+        """Get the input file contents for the calculation.
+
+        Returns:
+            tuple[str, str]: The contents of the control.in and geometry.in file
+        """
+        property_flags = {
+            "forces": "compute_forces",
+            "stress": "compute_analytical_stress",
+            "stresses": "compute_heat_flux",
+        }
+        updated_params = dict(**self._parameters)
+        for prop in self._properties:
+            aims_name = property_flags.get(prop, None)
+            if aims_name is not None:
+                updated_params[aims_name] = True
+
+        aims_geometry_in = AimsGeometryIn.from_structure(self._structure)
+        aims_control_in = AimsControlIn(updated_params)
+
+        os.makedirs(TMPDIR_NAME, exist_ok=True)
+        with monty.os.cd(TMPDIR_NAME):
+            aims_control_in.write_file(self._structure)
+            aims_control_in_content = AimsInputFile.from_file("control.in")
```

Comment:
@janosh This might be the reason for using `InputFiles` instead of just strings so there is a way to automatically read in a file as is. I can still remove it but the `from_file` calls from `InputFile` are convenient

Reply:
all that does is call `InputFile.from_file` which are just these 2 lines

```py
with zopen(filename, mode="rt") as f:
    return cls.from_str(f.read())
```

IMO doesn't deserve a whole wrapper class. might as well do

```py
with zopen(filename, mode="rt") as f:
    aims_control_in_content = f.read()
```

Reply:
True, for FHI-aims it is  not an issue. I will make the change today.